### PR TITLE
Fix legacy e2e tests that use library-provided handlers

### PR DESCRIPTION
**Pull request by andrewsg**
R: @mingzhaodotname CC: @dlorenc @isdal

### CODE REVIEW

File: multicore_runtime/legacy_e2e_support.py [hunk 1]
```diff
@@ -86,18 +86,23 @@ def legacy_app_for_script(script):
...
       runpy.run_module(modname, run_name='__main__')
+    except ImportError as e:
+      logging.exception('Error loading %s: %s', script, e)
     finally:
       revert_func()
     if app_holder:
       return app_holder[0]
     else:
-      logging.exception('Cannot load an app from %s' % script)
+      logging.exception('Cannot load an app from %s', script)
```
Comment (mingzhaodotname) [inline multicore_runtime/legacy_e2e_support.py]: we don't have an exception here, right?

File: multicore_runtime/legacy_e2e_support.py [hunk 2]
```diff
@@ -86,18 +86,23 @@ def legacy_app_for_script(script):
     The application as extracted from the script, or None.
   """
   with MockedWsgiHandler.python_25_app_lock:
-    modname, _ = script.rsplit('.', 1)
+    try:
+      modname, _ = script.rsplit('.', 1)
```
Comment (mingzhaodotname) [inline multicore_runtime/legacy_e2e_support.py]: how about modname = script.rsplit('.', 1)[0]

File: multicore_runtime/legacy_e2e_support.py [hunk 3]
```diff
@@ -86,18 +86,23 @@ def legacy_app_for_script(script):
...
+    except ValueError:
+      modname = script
     app_holder = []
     revert_func = lambda: None
     try:
       revert_func = stub_wsgi_utils(app_holder)
       runpy.run_module(modname, run_name='__main__')
+    except ImportError as e:
+      logging.exception('Error loading %s: %s', script, e)
```
Comment (mingzhaodotname) [inline multicore_runtime/legacy_e2e_support.py]: not a big thing. Just want to make sure that you are aware that logging.exception will print out the stack trace without giving e here. So, you can choose not to pass in e.

File: multicore_runtime/legacy_e2e_support.py [hunk 4]
```diff
@@ -54,7 +54,7 @@ def clean_up(self):
 
 
 def load_legacy_scripts_into_handlers(handlers):
-  """Preloads legacy CGI scripts.
+  """Preloads legacy CGI scripts. Static file handlers are not supported.
```
Comment (mingzhaodotname) [inline multicore_runtime/legacy_e2e_support.py]: do we need to update the test?

Comment (andrewsg) [inline multicore_runtime/legacy_e2e_support.py]: To reflect that static files are unsupported? No, legacy e2e tests are minimal but separate from the normal tests and all tests still pass.

File: multicore_runtime/legacy_e2e_support.py (hunk 3 above)
Comment (andrewsg) [inline multicore_runtime/legacy_e2e_support.py]: I wasn't aware of that; I will amend.

File: multicore_runtime/legacy_e2e_support.py (hunk 2 above)
Comment (andrewsg) [inline multicore_runtime/legacy_e2e_support.py]: Sure.

File: multicore_runtime/legacy_e2e_support.py (hunk 1 above)
Comment (andrewsg) [inline multicore_runtime/legacy_e2e_support.py]: That's right, I'll change it to logging.error

**Comment by andrewsg** [comment]
@mingzhaodotname PTAL

File: multicore_runtime/legacy_e2e_support.py [hunk 5]
```diff
@@ -86,18 +86,23 @@ def legacy_app_for_script(script):
     The application as extracted from the script, or None.
   """
   with MockedWsgiHandler.python_25_app_lock:
-    modname, _ = script.rsplit('.', 1)
+    try:
+      modname = script.rsplit('.', 1)[0]
```
Comment (mingzhaodotname) [inline multicore_runtime/legacy_e2e_support.py]: after this change, we don't need the "try except", right? unless you also want to catch the case that script is None

Comment (andrewsg) [inline multicore_runtime/legacy_e2e_support.py]: Oh, I see! Yes, I think you're right.

Comment (andrewsg) [inline multicore_runtime/legacy_e2e_support.py]: Fixed. PTAL

**Comment by mingzhaodotname** [comment]
LGTM

**Pull request merged**